client/context/load.go

```diff
@@ -0,0 +1,128 @@
+package context
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"github.com/docker/docker/pkg/contextstore"
+	"github.com/docker/go-connections/tlsconfig"
+	"github.com/pkg/errors"
+)
+
+// Context is a typed wrapper around a context-store context
+type Context struct {
+	Name          string
+	Host          string
+	SkipTLSVerify bool
+	APIVersion    string
+}
+
+// TLSData holds ca/cert/key raw data
+type TLSData struct {
+	CA   []byte
+	Key  []byte
+	Cert []byte
+}
+
+// LoadTLSData loads TLS materials for the context
+func (c *Context) LoadTLSData(s contextstore.Store) (*TLSData, error) {
+	tlsFiles, err := s.ListContextTLSFiles(c.Name)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve context tls info")
+	}
+	if epTLSFiles, ok := tlsFiles[dockerEndpointKey]; ok {
+		var keyBytes, certBytes, caBytes []byte
```

Comment:
nit: instead of using these 3 variables, you can directly instantiate a TLSData and set the fields in the `switch` ?